configuration/configurator/ConfigSchemaHelper.cpp

```diff
@@ -0,0 +1,889 @@
+            {
+                if (strXML.length() > 0 ? strcmp(strXML.str(), pXSDSchema) == 0 : true)
+                pSchema->dump(std::cout);
+            }
+        }
+    }
+}
+
+const char* CConfigSchemaHelper::printDocumentation(const char* comp)
+{
+    assert(comp != NULL && *comp != 0);
+    assert(m_pSchemaMapManager != NULL);
+
+    if (comp == NULL || *comp == 0)
+    {
+        return NULL;
+    }
+
+    CSchema* pSchema = NULL;
+
+
+    LOOP_THRU_BUILD_SET_MANAGER_BUILD_SET
+    {
+        const char *pSchemaName = CBuildSetManager::getInstance()->getBuildSetComponentFileName(idx);
+
+        if (pSchemaName != NULL && strcmp(comp, pSchemaName) == 0)
+        {
+             pSchema = m_pSchemaMapManager->getSchemaForXSD(pSchemaName);
+
+             assert(pSchema != NULL);
+
+             if (pSchema != NULL)
+             {
+                static StringBuffer strDoc;
+                strDoc.clear(); // needed when printing more than 1 component
+                pSchema->getDocumentation(strDoc);
+
+                return strDoc.str();
+             }
+        }
+    }
+
+    return NULL;
+}
+
+
+const char* CConfigSchemaHelper::printDojoJS(const char* comp)
+{
+    assert(comp != NULL && *comp != 0);
+    assert(m_pSchemaMapManager != NULL);
+
+    if (comp == NULL || *comp == 0)
+    {
+        return NULL;
+    }
+
+    CSchema* pSchema = NULL;
+
+    LOOP_THRU_BUILD_SET_MANAGER_BUILD_SET
+    {
+        const char *pSchemaName = CBuildSetManager::getInstance()->getBuildSetComponentFileName(idx);
+
+        if (pSchemaName != NULL && strcmp(comp, pSchemaName) == 0)
+        {
+             pSchema = m_pSchemaMapManager->getSchemaForXSD(pSchemaName);
+
+             assert(pSchema != NULL);
+
+             if (pSchema != NULL)
+             {
+                static StringBuffer strDoc;
+                pSchema->getDojoJS(strDoc);
+
+                return strDoc.str();
+             }
+        }
+    }
+
+    return NULL;
+}
+
+const char* CConfigSchemaHelper::printQML(const char* comp, int nIdx) const
+{
+    //assert(comp != NULL && *comp != 0);
+    if (! (comp != NULL && *comp != 0) )
+    {
+        DBGLOG("no component selected for QML, index = %d", nIdx);
+        return NULL;
+    }
+    assert(m_pSchemaMapManager != NULL);
+
+    static StringBuffer strQML;
+
+    strQML.clear();
+    resetTables();
+
+    if (comp == NULL || *comp == 0)
+    {
+        return NULL;
+    }
+
+    CSchema* pSchema = NULL;
+
+    LOOP_THRU_BUILD_SET_MANAGER_BUILD_SET
+    {
+        const char *pSchemaName = CBuildSetManager::getInstance()->getBuildSetComponentFileName(idx);
+
+        if (pSchemaName != NULL && strcmp(comp, pSchemaName) == 0)
+        {
+             pSchema = m_pSchemaMapManager->getSchemaForXSD(pSchemaName);
+
+             assert(pSchema != NULL);
+
+             if (pSchema != NULL)
+             {
+                 pSchema->getQML3(strQML, nIdx);
+                 return strQML.str();
```

Comment:
Ok, I see why it is static. But it's a terrible idea. Pass in a StringBuffer reference.